game/src/main/scala/io/aigar/game/Game.scala

```diff
@@ -43,32 +46,43 @@ class Game(val id: Int,
   val players = createPlayers
   val viruses = new Viruses(grid, math.max(Game.MinimumNumberOfPlayerModificator, playerIds.length))
   val resources = new Resources(grid)
+
   val startTime = Game.time
   var previousTime = 0f
   var currentTime = Game.MillisecondsPerTick / Game.MillisecondsPerSecond // avoid having an initial 0 delta time
   var tick = 0
 
-  def update: List[ScoreModification] = {
-    val deltaSeconds = currentTime - previousTime
-    var modifications = players.flatten {  _.update(deltaSeconds, grid, players) }
-    modifications :::= viruses.update(grid, players)
-    modifications :::= resources.update(grid, players)
-    tick += 1
+  def update: Future[(List[ScoreModification], serializable.GameState)] = {
+    Future {
+      this.synchronized {
+        blocking {
+          val deltaSeconds = currentTime - previousTime
+          var modifications = players.flatten {  _.update(deltaSeconds, grid, players) }
+          modifications :::= viruses.update(grid, players)
+          modifications :::= resources.update(grid, players)
+          tick += 1
 
-    previousTime = currentTime
-    currentTime = Game.time
-    modifications
+          previousTime = currentTime
+          currentTime = Game.time
+          (modifications, state)
```

Comment:
Is the `state` really implicit? I changed it like that, to be more atomic. Like, one the `update` is done, what's the state